src/clj_jargon/permissions.clj

```diff
@@ -153,65 +152,93 @@
      :write write
      :own   own}))
 
-(defn dataobject-perm?
+(defn- user-id-dataobject-perm?
+  "Utility function which checks if a given user-id has a given permission on
+  data-path"
+  [cm user-id data-path perm]
+  (let [results (dataobject-perms-rs cm user-id data-path)]
+    (some #(get (perm-map-for (first (.getColumnsAsList %))) perm) results)))
+
+(defn- loop-user-id-dataobject-perms
+  "Iterates through user IDs, checking if that user has the specified
+  permission or better on data-path, short-circuiting when true"
+  [cm user-ids data-path perm]
+  (when (seq user-ids)
+    (or (user-id-dataobject-perm? cm (first user-ids) data-path perm)
+        (recur cm (rest user-ids) data-path perm))))
```

Comment:
Could we use `some` with `user-id-dataobject-perm?` below as well, instead of this function that uses `recur`?
Since [some](https://clojuredocs.org/clojure.core/some) will also short-circuit, then maybe something like the following will work?
```clojure
(some #(user-id-dataobject-perm? cm % data-path checked-perm) (user-group-ids cm username))
```